src/main/kotlin/net/badgersmc/trivia/application/ContentFilter.kt

```
@@ -0,0 +1,45 @@
+import java.util.regex.Pattern
+import java.util.logging.Logger
+
+/**
+ * Filters trivia questions based on configurable regex patterns (REQ-009).
+ */
+class ContentFilter(private val config: ContentFilterConfig) {
+
+    private val blockedPatterns: List<Pattern> = config.blockedPatterns.mapNotNull { pattern ->
+        try {
+            Pattern.compile(pattern)
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)